src/client.rs

```diff
@@ -137,6 +140,28 @@ impl Client {
             .await?)
     }
 
+    /// This potentially deletes a resource _if it is not already deleted_.
+    /// It checks by looking at the `deletion_timestamp` of the resource.
+    /// If that is the case it'll return a `Ok(None)`.
+    pub async fn maybe_delete<T>(&self, resource: &T) -> OperatorResult<Option<Either<T, Status>>>
```

Comment:
I don't like "maybe": The outcome is a deleted resource.
What about delete_dangling or delete_safely (in terms of we don't delete again if it is already up for deletion?)

Reply:
Yeah, that's a valid point.... I'll think about it. Thanks!

Reply:
@soenkeliebau pinging to see if you have any ideas for a better name

Reply:
no definitive ideas ... something like "ensure_deleted"  / "delete_if_no_deletion_timestamp" ...

That being said, is there a specific reason to make this a separate function, when would we use `delete` over this one? To my mind the behavior in this function should be the default behavior in delete. Could we maybe change the return value of delete to include the deletion timestamp and a flag if it was already present or something similar?